common/unsafe/src/main/java/org/apache/spark/sql/catalyst/util/CollationFactory.java

```diff
@@ -88,12 +90,45 @@ public Optional<String> getVersion() {
     }
   }
 
+  public record CollationMeta(
+    String catalog,
+    String schema,
+    String collationName,
+    String language,
+    String country,
+    String icuVersion,
+    String padAttribute,
+    boolean accentSensitivity,
+    boolean caseSensitivity) {
+
+    public static CollationMeta fromCollation(
+        String catalog, String schema, Collation collation) {
+      return new CollationMeta(
+        catalog,
+        schema,
+        collation.collationName,
+        collation.language,
+        collation.country,
+        collation.icuVersion,
+        collation.padAttribute,
+        collation.accentSensitivity,
+        collation.caseSensitivity);
+    }
+  }
+
   /**
    * Entry encapsulating all information about a collation.
    */
   public static class Collation {
+
     public final String collationName;
     public final String provider;
+    public final String language;
+    public final String country;
+    public final String icuVersion;
+    public final String padAttribute;
+    public final boolean accentSensitivity;
+    public final boolean caseSensitivity;
```

Comment:
I would say let's not add this information to Collation object, this object is heavy with Comparator and Collator in itself, so lets try to keep CollationMeta built from the minimal amount of information we need, that would be CollationIdentifier which has the name, provider and version. Building Collation object is expensive and will store all Collation objects to the map on a single call of SHOW COLLATIONS, so let's keep it simple and efficient.

Reply:
There is a problem with this. If we do not include this information in the `Collation object`, we will need to add a new method `buildCollationMeta` below, otherwise we will not be able to retrieve this data from `CollationFactory`
This seems to go back to the implementation of the previous version, do you think this is okay?

Reply:
<img width="538" alt="image" src="https://github.com/user-attachments/assets/bde62873-d765-41eb-93b4-54a4757ad3b6">

Reply:
Yeah, I would say this is better than building Collation objects. But we can keep the filtering inside of SessionCatalog so we do not have to reimplement filter again, but use StringUtils functionality.

Reply:
Okay, it's a pleasant agreement. Let me update it now, thanks!